src/performance/metrics/StatisticCollector.cpp

```diff
@@ -317,6 +319,884 @@ double StatisticCollector::getLoadAverage() {
+            // Skip loop devices and ram devices
+            if (strncmp(device, "loop", 4) != 0 && strncmp(device, "ram", 3) != 0) {
+                secondReading[device] = std::make_pair(sectors_read, sectors_written);
+            }
+        }
+    }
+    fclose(file2);
+    
+    // Record end time
+    clock_gettime(CLOCK_MONOTONIC, &endTime);
+    
+    // Calculate elapsed time in seconds
+    double elapsedTime = (endTime.tv_sec - startTime.tv_sec) + 
+                        (endTime.tv_nsec - startTime.tv_nsec) / 1000000000.0;
+    
+    if (elapsedTime <= 0.0) {
+        stat_logger.error("Invalid elapsed time for disk read/write calculation");
+        return diskRates;
+    }
+    
+    // Calculate read/write KB per second for each device
+    for (const auto& entry : firstReading) {
+        const std::string& device = entry.first;
+        unsigned long long firstSectorsRead = entry.second.first;
+        unsigned long long firstSectorsWritten = entry.second.second;
+        
+        if (secondReading.find(device) != secondReading.end()) {
+            unsigned long long secondSectorsRead = secondReading[device].first;
+            unsigned long long secondSectorsWritten = secondReading[device].second;
+            
+            // Calculate the deltas (handling potential counter wraparound)
+            unsigned long long deltaSectorsRead, deltaSectorsWritten;
+            
+            if (secondSectorsRead >= firstSectorsRead) {
+                deltaSectorsRead = secondSectorsRead - firstSectorsRead;
+            } else {
+                // Counter wrapped around, assume it's a 64-bit counter
+                deltaSectorsRead = (ULLONG_MAX - firstSectorsRead) + secondSectorsRead + 1;
+            }
+            
+            if (secondSectorsWritten >= firstSectorsWritten) {
+                deltaSectorsWritten = secondSectorsWritten - firstSectorsWritten;
+            } else {
+                // Counter wrapped around, assume it's a 64-bit counter
+                deltaSectorsWritten = (ULLONG_MAX - firstSectorsWritten) + secondSectorsWritten + 1;
+            }
+            
+            // Convert sectors to KB: sectors are 512 bytes, so divide by 2 to get KB
+            // Then divide by elapsed time to get KB per second
+            double readKBPerSecond = (static_cast<double>(deltaSectorsRead) / 2.0) / elapsedTime;
+            double writeKBPerSecond = (static_cast<double>(deltaSectorsWritten) / 2.0) / elapsedTime;
+            
+            diskRates[device] = std::make_pair(readKBPerSecond, writeKBPerSecond);
+        }
+    }
+    
+    return diskRates;
+}
+
+std::map<std::string, double> StatisticCollector::getDiskBlockSizeKB() {
+    std::map<std::string, double> diskBlockSizes;
+    
+    FILE *file = fopen("/proc/diskstats", "r");
+    if (!file) {
+        stat_logger.error("Cannot open /proc/diskstats for disk block size reading");
+        return diskBlockSizes;
+    }
+    
+    char line[256];
```

Comment:
Why do we have different line size of 256 here. We had above 1024 line sizes. Also can we have this move to a constant?